ledger/acctupdates.go

```diff
@@ -275,6 +275,22 @@ func (au *accountUpdates) close() {
 	<-au.commitSyncerClosed
 }
 
+// IsWritingCatchpointFile returns true when a catchpoint file is being generated. The function is used by the catchup service so
+// that the memory pressure could be decreased until the catchpoint file writing is complete.
+func (au *accountUpdates) IsWritingCatchpointFile() bool {
+	au.accountsMu.RLock()
+	defer au.accountsMu.RUnlock()
+	// if we're still writing the previous balances, we can't move forward yet.
+	select {
+	case <-au.catchpointWriting:
```

Comment:
Out of curiosity, why is `au.catchpointWriting` a channel and not a boolean variable protected by `accountsMu`?

Reply:
It would probably be a good idea to use a different synchronization primitive at this point, but I didn't want to make such a big change for this fix.